src/ring_theory/ideal/operations.lean

```diff
@@ -1139,6 +1139,29 @@ end
 
 section quotient_algebra
 
+/-- The `R`-algebra structure on `R/I` -/
+instance {I : ideal R} : algebra R (ideal.quotient I) := (ideal.quotient.mk I).to_algebra
+
+/-- The canonical morphism `R →ₐ[R] I.quotient`, for `I` an ideal of `R`, as morphism of
+`R`-algebras. -/
+def quotient.mk.alg (I : ideal R) : R →ₐ[R] I.quotient :=
+⟨λ a, submodule.quotient.mk a, rfl, λ _ _, rfl, rfl, λ _ _, rfl, λ _, rfl⟩
+
+lemma quotient.mk.alg_to_ring_hom (I : ideal R) :
+  (quotient.mk.alg I).to_ring_hom = ideal.quotient.mk I := rfl
+
+@[simp] lemma quotient.mk.alg_eq_mk (I : ideal R) :
+  ⇑(quotient.mk.alg I) = ideal.quotient.mk I := rfl
+
+/-- The canonical morphism `R →ₐ[R] I.quotient` is surjective. -/
+lemma quotient.mk.alg_surjective (I : ideal R) : function.surjective (quotient.mk.alg I) :=
+λ y, quotient.induction_on' y (λ x, exists.intro x rfl)
+
+/-- The kernel of `R →ₐ[R] I.quotient` is `I`. -/
+@[simp]
+lemma quotient.mk.alg_ker (I : ideal R) : (quotient.mk.alg I).to_ring_hom.ker = I :=
+by rw [quotient.mk.alg_to_ring_hom _, ideal.mk_ker]
```

Comment:
Does this work as a proof?
```suggestion
ideal.mk_ker I
```

Reply:
No :(

`function expected at  mk_ker term has type (quotient.mk ?m_3).ker = ?m_3`

Reply:
Ok, what if you do that but without the `I`?

Reply:
Ah, of course! 🤦‍♂️